src/Translator/src/FileDumper/AssetMapperTranslationsFileDumper.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Translator\FileDumper;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Translation\MessageCatalogueInterface;
+
+/**
+ * @author Maelan Le Borgne <maelan.leborgne@gmail.com>
+ *
+ * @final
+ *
+ * @experimental
+ *
+ * @phpstan-type Domain string
+ * @phpstan-type Locale string
+ * @phpstan-type MessageId string
+ */
+class AssetMapperTranslationsFileDumper implements TranslationFileDumperInterface
+{
+    private array $excludedDomains = [];
+    private array $domains = [];
+
+    public function __construct(
+        private string $dumpDir,
+        private Filesystem $filesystem,
+    ) {
+    }
+
+    public function dump(MessageCatalogueInterface ...$catalogues): void
+    {
+        $this->filesystem->mkdir($this->dumpDir);
+        $this->filesystem->remove($this->dumpDir.'/domains');
+        $this->filesystem->mkdir($this->dumpDir.'/domains');
+
+        $domainSplit = [];
+        foreach ($this->getTranslations(...$catalogues) as $translationId => $translationDetails) {
+            $domain = $translationDetails['domain'];
+            $locale = $translationDetails['locale'];
+            $domainFileName = str_ends_with($domain, MessageCatalogueInterface::INTL_DOMAIN_SUFFIX)
+                ? substr($domain, 0, -\strlen(MessageCatalogueInterface::INTL_DOMAIN_SUFFIX))
+                : $domain;
+
+            $domainSplit[$domainFileName] ??= [];
+            $domainSplit[$domainFileName][$translationId] ??= ['id' => $translationId, 'translations' => []];
+            $domainSplit[$domainFileName][$translationId]['translations'][$domain] ??= [];
+            $domainSplit[$domainFileName][$translationId]['translations'][$domain][$locale] = $translationDetails['message'];
```

Comment:
So you'd keep all the locale in one file ?

Reply:
Yes that's the plan right now, otherwise we'd have to import to many modules (1 per combination of domain/locale). Though a filter on the dumped locales could help remove the unnecessary ones. Unless you have somthing else in mind ?